src/tests/ftest/util/apricot/apricot/test.py

```diff
@@ -132,17 +130,87 @@ def __init__(self, *args, **kwargs):
         # use 'add_cancel_ticket(<ticket>)' to add to this set.
         self._teardown_cancel = set()
         self._teardown_errors = []
+        self.basepath = None
+        self.prefix = None
+        self.ofi_prefix = None
 
     def setUp(self):
         """Set up each test case."""
+        # get paths from the build_vars generated by build
+        with open('../../.build_vars.json') as build_vars:
+            build_paths = json.load(build_vars)
+        self.basepath = os.path.normpath(os.path.join(build_paths['PREFIX'],
+                                                      '..') + os.path.sep)
+        self.prefix = build_paths['PREFIX']
+        try:
+            self.ofi_prefix = build_paths['OFI_PREFIX']
+        except KeyError:
+            self.ofi_prefix = "/usr"
+        self.skip_from_list()
         self.check_variant_skip()
         self.log.info("*** SETUP running on %s ***", str(detect()))
         super(Test, self).setUp()
 
-    def check_variant_skip(self):
+    def skip_from_list(self):
+        """Check if test is in skip list"""
+        skip_list = ""
+        try:
+            skip_list = open("/scratch/CI-skip-list").read()
+        except Exception as excpt: # pylint: disable=broad-except
+            print("Unable to read skip list: ", excpt)
+            print("Trudging on without skipping known failing tests")
+            return
+
+        for item in skip_list.splitlines():
+            vals = item.split('|')
+            skip_it, ticket = self._check_variant_skip(eval(vals[0])) # pylint: disable=eval-used
+            if skip_it:
+                # test is on the skiplist
+                if len(vals) > 1:
+                    # but there is a commit that fixes it
+                    commits = None
+                    try:
+                        commits = open('/tmp/commit_list').read().splitlines()
+                    except Exception as excpt: # pylint: disable=broad-except
+                        print("Unable to read commit list: ", excpt)
+                        print("Trudging on without skipping known failing "
+                              "tests")
+                    if commits and vals[1] in commits:
+                        # fix is in this code base
+                        print("On the skip list for ticket {}, but is "
+                              "fixed in {} so not "
+                              "skipping".format(ticket, vals[1]))
+                    else:
+                        # fix is not in this code base
+                        self.cancel("Skipping due to being on the "
```

Comment:
Should this just be a `self.log.info(<message>)` + `self.cancelForTicket(ticket)`?

Reply:
Perhaps.